code/game/machinery/computer/dropship_weapons.dm

```diff
@@ -196,497 +123,660 @@
-		var/mob/M = usr
-		if(!skillcheck(M, SKILL_PILOT, SKILL_PILOT_TRAINED)) //only pilots can fire dropship weapons.
-			to_chat(usr, SPAN_WARNING("A screen with graphics and walls of physics and engineering values open, you immediately force it closed."))
-			return
+		if("execute_simulated_firemission")
+			if(!configuration)
+				to_chat(user, SPAN_WARNING("No configured firemission"))
+				return
+			simulate_firemission(user)
+			. = TRUE
 
-		var/chosen = stripped_input(usr, "Select Fire Mission length, from 0 to [firemission_envelope.max_offset]", "Select Offset", "[firemission_envelope.recorded_offset]", 2)
-		var/chosen_offset = text2num(chosen)
+		if("switch_firemission")
+			configuration = tgui_input_list(user, "Select firemission to simulate", "Select firemission", firemission_envelope.missions, 30 SECONDS)
+			if(!selected_firemission)
+				to_chat(user, SPAN_WARNING("No configured firemission"))
+				return
+			if(!configuration)
+				configuration = selected_firemission
+			. = TRUE
 
-		if(chosen_offset == null)
-			to_chat(usr, SPAN_WARNING("Error with offset detected."))
-			return
+		if("switchmode")
+			simulation.dummy_mode = tgui_input_list(user, "Select target type to simulate", "Target type", simulation.target_types, 30 SECONDS)
+			if(!simulation.dummy_mode)
+				simulation.dummy_mode = CLF_MODE
+			. = TRUE
 
-		update_offset(chosen_offset)
+		if("set-camera")
+			var/target_camera = params["equipment_id"]
+			set_camera_target(target_camera)
+			return TRUE
+
+		if("set-camera-sentry")
+			var/equipment_tag = params["equipment_id"]
+			for(var/obj/structure/dropship_equipment/equipment as anything in shuttle.equipments)
+				var/mount_point = equipment.ship_base.attach_id
+				if(mount_point != equipment_tag)
+					continue
+				if (istype(equipment, /obj/structure/dropship_equipment/sentry_holder))
+					var/obj/structure/dropship_equipment/sentry_holder/sentry = equipment
+					var/obj/structure/machinery/defenses/sentry/defense = sentry.deployed_turret
+					if (defense.has_camera)
+						defense.set_range()
+						var/datum/shape/rectangle/current_bb = defense.range_bounds
+						SEND_SIGNAL(src, COMSIG_CAMERA_SET_AREA, current_bb.center_x, current_bb.center_y, defense.loc.z, current_bb.width, current_bb.height)
+				return TRUE
+
+		if("clear-camera")
+			set_camera_target(null)
+			return TRUE
+
+		if("medevac-target")
+			var/equipment_tag = params["equipment_id"]
+			for(var/obj/structure/dropship_equipment/equipment as anything in shuttle.equipments)
+				var/mount_point = equipment.ship_base.attach_id
+				if(mount_point != equipment_tag)
+					continue
+				if (istype(equipment, /obj/structure/dropship_equipment/medevac_system))
+					var/obj/structure/dropship_equipment/medevac_system/medevac = equipment
+					var/target_ref = params["ref"]
+					medevac.automate_interact(user, target_ref)
+					if(medevac.linked_stretcher)
+						SEND_SIGNAL(src, COMSIG_CAMERA_SET_TARGET, medevac.linked_stretcher, 5, 5)
```

Comment:
This seems to be causing a runtime at the moment due to the medevac stretcher being passed as the `target` argument, rather than a camera.
The signal calls `set_camera()`, which calls `update_target_camera()`, which then fails here because the stretcher doesn't have a `can_use()` proc:
https://github.com/cmss13-devs/cmss13/blob/dec1a8429726bc70e4dd8598d38da8d694b0611a/code/game/camera_manager/camera_manager.dm#L159-L161
```
[2023-12-23 12:55:22.339] runtime error: undefined proc or verb /obj/structure/bed/medevac_stretcher/can use().
 - 
 - proc name: update target camera (/datum/component/camera_manager/proc/update_target_camera)
 -   source file: camera_manager.dm,161
 -   usr: Jermaine Juarez (/mob/living/carbon/human)
 -   src: /datum/component/camera_manage... (/datum/component/camera_manager)
 -   usr.loc: the floor (272,33,4) (/turf/open/shuttle/dropship)
 -   call stack:
 - /datum/component/camera_manage... (/datum/component/camera_manager): update target camera()
 - /datum/component/camera_manage... (/datum/component/camera_manager): set camera(the \'Normandy\' weapons contr... (/obj/structure/machinery/computer/dropship_weapons/dropship2), the medevac stretcher (/obj/structure/bed/medevac_stretcher), 5, 5)
 - CallAsync(/datum/component/camera_manage... (/datum/component/camera_manager), /datum/component/camera_manage... (/datum/component/camera_manager/proc/set_camera), /list (/list))
 - the \'Normandy\' weapons contr... (/obj/structure/machinery/computer/dropship_weapons/dropship2):  SendSignal("camera_manager_set_target", /list (/list))
 - the \'Normandy\' weapons contr... (/obj/structure/machinery/computer/dropship_weapons/dropship2): ui act("medevac-target", /list (/list), /datum/tgui (/datum/tgui), /datum/ui_state/not_incapacita... (/datum/ui_state/not_incapacitated_and_adjacent_strict_state))
 - /datum/tgui (/datum/tgui): on act message("medevac-target", /list (/list), /datum/ui_state/not_incapacita... (/datum/ui_state/not_incapacitated_and_adjacent_strict_state))
 - /datum/tgui (/datum/tgui): on message("act/medevac-target", /list (/list), /list (/list))
 - /datum/tgui_window (/datum/tgui_window): on message("act/medevac-target", /list (/list), /list (/list))
 - tgui Topic(/list (/list))
 - SabreML (/client): Topic("type=act%2Fmedevac-target&payl...", /list (/list), null)
```

Reply:
Thanks for the report - I've resolved this